modules/activemq-artemis-launch/added/drain.sh

```diff
@@ -1,177 +1,244 @@
 #!/bin/sh
 
-export BROKER_HOST=`hostname -f`
+function log() {
+  logtime=$(date)
+  echo "[$logtime]-[drain.sh] $1"
+}
 
-echo "[drain.sh] drainer container ip(from hostname) is $BROKER_HOST"
+function get_total_messages_on_broker() {
 
-instanceDir="${HOME}/${AMQ_NAME}"
+  log "get total messages on broker ${BROKER_HOST}"
 
-ENDPOINT_NAME="${AMQ_NAME}-amq-headless"
+  RET_VALUE=$(curl -s -G -k http://${AMQ_USER}:${AMQ_PASSWORD}@${BROKER_HOST}:8161/console/jolokia/read/org.apache.activemq.artemis:broker=%22${AMQ_NAME}%22/AddressNames)
 
-if [ "$HEADLESS_SVC_NAME" ]; then
-  ENDPOINT_NAME=$HEADLESS_SVC_NAME
-fi
+  PYCMD=$(cat <<EOF
+import sys, json
+addrs = ''
+value = json.load(sys.stdin)['value']
+for addr in value:
+    addrs = addrs + ' ' + addr
+print addrs
+EOF
+)
 
-endpointsUrl="https://${KUBERNETES_SERVICE_HOST:-kubernetes.default.svc}:${KUBERNETES_SERVICE_PORT:-443}/api/v1/namespaces/${POD_NAMESPACE}/"
-endpointsAuth="Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)"
+  all_addresses=$(echo "$RET_VALUE" | python2 -c "$PYCMD")
+  log "get back all addresses $all_addresses"
+  arr=($all_addresses)
+
+  TOTAL_MESSAGES_ON_BROKER=0
+  for address in "${arr[@]}"
+  do
+    log "checking on address ${address}"
+    M_COUNT=$(curl -s -G -k http://${AMQ_USER}:${AMQ_PASSWORD}@${BROKER_HOST}:8161/console/jolokia/read/org.apache.activemq.artemis:broker=%22${AMQ_NAME}%22,address=%22${address}%22,component=addresses/MessageCount)
+    value=$(echo $M_COUNT | python2 -c "import sys, json; print json.load(sys.stdin)['value']")
```

Comment:
Installing jq or yq the python scripts could be replaced by simpler commands, i.e.
```suggestion
    value=$(echo $M_COUNT | jq -r '.value')
```

Reply:
will do.